src/wasm.cc

```diff
@@ -394,7 +394,10 @@ void WasmBase::startVm(ContextBase *root_context) {
              // time
              "wasi_unstable.clock_time_get", "wasi_snapshot_preview1.clock_time_get",
              // random
-             "wasi_unstable.random_get", "wasi_snapshot_preview1.random_get"});
+             "wasi_unstable.random_get", "wasi_snapshot_preview1.random_get",
+             // For proxy-wasm-go-sdk Go runtime initialization
+             "wasi_unstable.fd_fdstat_get", "wasi_snapshot_preview1.fd_fdstat_get",
+             "wasi_unstable.fd_fdstat_set_flags", "wasi_snapshot_preview1.fd_fdstat_set_flags"});
```

Comment:
Mostly for the sake of my own curiosity, but do you happen to have a trace (Envoy should print it at the `trace` log level) from a sample Go SDK plugin?

Reply:
Is this what you're interested in?

```
TRACE from integration: [host->vm] _initialize()                                                                                                                                        
TRACE from integration: [vm->host] wasi_snapshot_preview1.clock_time_get(1, 0, 1599104)                                                                                                 
TRACE from integration: [vm<-host] wasi_snapshot_preview1.clock_time_get return: 0                                                                                                      
TRACE from integration: [vm->host] wasi_snapshot_preview1.clock_time_get(1, 0, 1599104)                                                                                                 
TRACE from integration: [vm<-host] wasi_snapshot_preview1.clock_time_get return: 0                                                                                                      
TRACE from integration: [vm->host] wasi_snapshot_preview1.random_get(1572840, 32)                                                                                                       
TRACE from integration: [vm<-host] wasi_snapshot_preview1.random_get return: 0                                                                                                          
TRACE from integration: [vm->host] wasi_snapshot_preview1.args_sizes_get(1599060, 1599056)                                                                                              
TRACE from integration: [vm<-host] wasi_snapshot_preview1.args_sizes_get return: 0                                                                                                      
TRACE from integration: [vm->host] wasi_snapshot_preview1.environ_sizes_get(1599048, 1599052)                                                                                           
TRACE from integration: [vm<-host] wasi_snapshot_preview1.environ_sizes_get return: 0                                                                                                   
TRACE from integration: [vm->host] wasi_snapshot_preview1.clock_time_get(1, 0, 1599120)                                                                                                 
TRACE from integration: [vm<-host] wasi_snapshot_preview1.clock_time_get return: 0                                                                                                      
TRACE from integration: [vm->host] wasi_snapshot_preview1.clock_time_get(1, 0, 1598856)                                                                                                 
TRACE from integration: [vm<-host] wasi_snapshot_preview1.clock_time_get return: 0                                                                                                      
TRACE from integration: [vm->host] wasi_snapshot_preview1.clock_time_get(1, 0, 4384600)                                                                                                 
TRACE from integration: [vm<-host] wasi_snapshot_preview1.clock_time_get return: 0                                                                                                      
TRACE from integration: [vm->host] wasi_snapshot_preview1.clock_time_get(1, 0, 1598880)                                                                                                 
TRACE from integration: [vm<-host] wasi_snapshot_preview1.clock_time_get return: 0                                                                                                      
TRACE from integration: [vm->host] wasi_snapshot_preview1.clock_time_get(1, 0, 1598832)                                                                                                 
TRACE from integration: [vm<-host] wasi_snapshot_preview1.clock_time_get return: 0                                                                                                      
TRACE from integration: [vm->host] wasi_snapshot_preview1.fd_fdstat_get(0, 4451768)                                                                                                     
TRACE from integration: [vm<-host] wasi_snapshot_preview1.fd_fdstat_get return: 8                                                                                                       
TRACE from integration: [vm->host] wasi_snapshot_preview1.fd_fdstat_get(1, 4451768)                                                                                                     
TRACE from integration: [vm<-host] wasi_snapshot_preview1.fd_fdstat_get return: 0                                                                                                       
TRACE from integration: [vm->host] wasi_snapshot_preview1.fd_fdstat_set_flags(1, 4)                                                                                                     
TRACE from integration: [vm<-host] wasi_snapshot_preview1.fd_fdstat_set_flags return: 52                                                                                                
TRACE from integration: [vm->host] wasi_snapshot_preview1.fd_fdstat_get(2, 4451768)                                                                                                     
TRACE from integration: [vm<-host] wasi_snapshot_preview1.fd_fdstat_get return: 0                                                                                                       
TRACE from integration: [vm->host] wasi_snapshot_preview1.fd_fdstat_set_flags(2, 4)                                                                                                     
TRACE from integration: [vm<-host] wasi_snapshot_preview1.fd_fdstat_set_flags return: 52
TRACE from integration: [vm->host] wasi_snapshot_preview1.fd_prestat_get(3, 4451804)   
TRACE from integration: [vm<-host] wasi_snapshot_preview1.fd_prestat_get return: 8
TRACE from integration: [vm->host] wasi_snapshot_preview1.clock_time_get(1, 0, 4451832)
TRACE from integration: [vm<-host] wasi_snapshot_preview1.clock_time_get return: 0
TRACE from integration: [vm->host] wasi_snapshot_preview1.fd_fdstat_get(0, 4451712)       
TRACE from integration: [vm<-host] wasi_snapshot_preview1.fd_fdstat_get return: 8 
TRACE from integration: [vm->host] wasi_snapshot_preview1.fd_fdstat_get(1, 4451712)                                                                                                     
TRACE from integration: [vm<-host] wasi_snapshot_preview1.fd_fdstat_get return: 0    
TRACE from integration: [vm->host] wasi_snapshot_preview1.fd_fdstat_get(2, 4451712)    
TRACE from integration: [vm<-host] wasi_snapshot_preview1.fd_fdstat_get return: 0 
TRACE from integration: [host<-vm] _initialize return: void     
```

Reply:
Yes, thank you.